### ISSUE TITLE
setImageLayout() uses invalid stages for pipeline barrier

### ISSUE THREAD

**Issue by rafalcieslak** [nvpro-pipeline/VkHLF#31]
[Specs for vkCmdPipelineBarrier](https://www.khronos.org/registry/vulkan/specs/1.0/html/vkspec.html#VUID-vkCmdPipelineBarrier-pMemoryBarriers-01184) state that:

> Any given element of pMemoryBarriers, pBufferMemoryBarriers or pImageMemoryBarriers must not have any access flag included in its srcAccessMask member if that bit is not supported by any of the pipeline stages in srcStageMask, as specified in the table of supported access types.

According to [the table](https://www.khronos.org/registry/vulkan/specs/1.0/html/vkspec.html#synchronization-access-types-supported) in Vulkan specs `eTopOfPipe` is not supported by any access flags.

But [at the moment, setImageLayout](https://github.com/nvpro-pipeline/VkHLF/blob/3b1be2b8aa057fa1c09361c1322ad63410a788be/vkhlf/src/CommandBuffer.cpp#L698) prepares pipeline barrier with these invalid arguments:

```
commandBuffer->pipelineBarrier(vk::PipelineStageFlagBits::eTopOfPipe, vk::PipelineStageFlagBits::eTopOfPipe, {}, nullptr, nullptr, imageMemoryBarrier);
```

Standard validation layers detect this issue on any call to `setImageLayout`.

**Comment by playmer** [comment]
Does anyone have a fix for this? I've just updated to 1.66.1 and noticed the validation layers going off. I'm not too familiar with some of the Pipeline stages and such, but I took a small crack at it here:

https://github.com/playmer/VkHLF/commit/1577f20b10af9cdba10f586819b3d43d988f3e1e

I mainly cross referenced the spec:

https://www.khronos.org/registry/vulkan/specs/1.0/html/vkspec.html#synchronization-dependencies

At 6.1.3 table 4 to generate the flags and it doesn't seem to complain, but I'm also not using too many different image types.

Any thoughts on something like this? Or something better?

@asuessenbach, @mtavenrath?
